assets/javascripts/discourse/components/chat-toggle-membership-btn.js

```diff
@@ -0,0 +1,91 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import ChatApi from "discourse/plugins/discourse-chat/discourse/lib/chat-api";
+import { action, computed } from "@ember/object";
+
+export default class ChatToggleMembershipBtn extends Component {
```

Comment:
we should try glimmer component if possible here

Reply:
Yeah still wasn't sure whether I should, I will try it out :+1:

Reply:
As discussed can't do this because then that would require the ChatChannel model to become a native class...and then lots of other things need to change